Microsoft.Toolkit.Uwp.UI.Controls/MarkdownTextBlock/Helpers/Common.cs

```diff
@@ -0,0 +1,623 @@
+                    StartOfNextLine = startOfNextLine,
+                };
+
+                if (nonSpaceChar == '\0')
+                {
+                    // The line is empty or nothing but whitespace.
+                    lineStartsNewParagraph = true;
+                }
+
+                // Repeat.
+                pos = startOfNextLine;
+            }
+        }
+
+        /// <summary>
+        /// Skips a certain number of quote characters (>).
+        /// </summary>
+        /// <returns>Skip Quote Chars</returns>
+        public static int SkipQuoteCharacters(string markdown, int start, int end, int quoteDepth)
+        {
+            if (quoteDepth == 0)
+            {
+                return start;
+            }
+
+            int startOfLine = start;
+            int nonSpacePos = start;
+            char nonSpaceChar = '\0';
+
+            while (true)
+            {
+                // Find the next non-space char.
+                while (nonSpacePos < end)
+                {
+                    char c = markdown[nonSpacePos];
+                    if (c == '\r' || c == '\n')
+                    {
+                        // The line is either entirely whitespace, or is empty.
+                        break;
+                    }
+
+                    if (c != ' ' && c != '\t')
+                    {
+                        // The line has content.
+                        nonSpaceChar = c;
+                        break;
+                    }
+
+                    nonSpacePos++;
+                }
+
+                // When parsing blocks in a blockquote context, we need to count the number of
+                // quote characters ('>').  If there are less than expected AND this is the
+                // start of a new paragraph, then stop parsing.
+                if (quoteDepth == 0)
+                {
+                    break;
+                }
+
+                if (nonSpaceChar == '>')
+                {
+                    // Expected block quote characters should be ignored.
+                    quoteDepth--;
+                    nonSpacePos++;
+                    nonSpaceChar = '\0';
+                    startOfLine = nonSpacePos;
+
+                    // Ignore the first space after the quote character, if there is one.
+                    if (startOfLine < end && markdown[startOfLine] == ' ')
+                    {
+                        startOfLine++;
+                        nonSpacePos++;
+                    }
+                }
+                else
+                {
+                    // There were less block quote characters than expected.
+                    break;
+                }
+            }
+
+            return startOfLine;
+        }
+
+        /// <summary>
+        /// Determines if a character is a whitespace character.
+        /// </summary>
+        /// <returns>true if is white space</returns>
+        public static bool IsWhiteSpace(char c)
```

Comment:
I'm going to keep this for now, the standard for IsWhiteSpace seems to include a lot more chars than what we check for here, and since markdown is quite strick on white space I don't want to mess with what we have.